### PR TITLE
Reveal the taskbar when the user has the Terminal maximized or fullscreen

### LINKED ISSUE #1438: Cannot access taskbar when terminal is maximized/fullscreen
# Environment - Windows 10 Pro Insider Preview Build 18922

```none
Windows build number: 10.0.18922.1000
Windows Terminal version (if applicable):

Any other software? No
```

# Steps to reproduce - 
1. Have your taskbar set to auto-hide.
2. Start (amazing) new windows terminal.
3. Maximize terminal.
4. Move the mouse to the bottom of the screen trying to get the hidden taskbar to pop up.

# Expected behavior - 
Hidden taskbar should still pop up even though the terminal is maximized

# Actual behavior - 
Cannot access the taskbar with the mouse when the terminal is maximized

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

When we are maximized or fullscreened, check for the presence of the taskbar in auto-hide mode. If the Terminal finds the taskbar on any side of the monitor, adjust our window rect by just a little bit, so that the taskbar can still be revealed by the user mousing over that edge.

## PR Checklist
* [x] Closes #1438
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments

Note to future code archeologists:
This doesn't seem to work for fullscreen on the primary display. However, testing a bunch of other apps with fullscreen modes and an auto-hiding taskbar has shown that _none_ of them reveal the taskbar from fullscreen mode. This includes Edge, Firefox, Chrome, Sublime Text, Powerpoint - none seemed to support this. 

This does however work fine for maximized.

## Validation Steps Performed

I'm maximized and fullscreened the Terminal a lot in the last two days.

### CODE REVIEW

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp [hunk 1]
```diff
@@ -332,38 +334,101 @@ int NonClientIslandWindow::_GetResizeHandleHeight() const noexcept
...
+            // Note to future code archeologists:
+            // This doesn't seem to work for fullscreen on the primary display.
+            // However, testing a bunch of other apps with fullscreen modes
+            // and an auto-hiding taskbar has shown that _none_ of them
+            // reveal the taskbar from fullscreen mode. This includes Edge,
+            // Firefox, Chrome, Sublime Text, Powerpoint - none seemed to
+            // support this.
+            //
+            // This does however work fine for maximized.
+            if (onTop)
```
Comment (leonMSFT) [inline src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: nit: a taskbar can only be on one side of a monitor at a time right? then these if-statements could turn into if-elses

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp [hunk 2]
```diff
@@ -332,38 +334,101 @@ int NonClientIslandWindow::_GetResizeHandleHeight() const noexcept
...
+
+        // First, check if we have an auto-hide taskbar at all:
+        APPBARDATA autohide{ 0 };
+        autohide.cbSize = sizeof(autohide);
+        UINT state = (UINT)SHAppBarMessage(ABM_GETSTATE, &autohide);
+        if (WI_IsFlagSet(state, ABS_AUTOHIDE))
+        {
+            // This helper can be used to determine if there's a auto-hide
+            // taskbar on the given edge of the monitor we're currently on.
+            auto hasAutohideTaksbar = [&monInfo](const UINT edge) -> bool {
```
Comment (leonMSFT) [inline src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: nit: `hasAutohideTaskbar`

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: Actually, _no_. It appears that apps can use this API to add additional bars to other sides of the display? It's really unclear, this whole API is a disaster -.-

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: fak

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp (hunk 1 above)
Comment (leonMSFT) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: wtf yikes 😨

**Pull request merged**